deploy-tool/cert/cert.go

```diff
@@ -0,0 +1,95 @@
+package cert
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/x509"
+	"deploy-tool/dns"
+	"fmt"
+	"github.com/caddyserver/certmagic"
+	"os"
+	"strings"
+)
+
+// IssueCertificate asks the Let's Encrypt CA to validate and sign a certificate for a site using the DNS01 strategy
+// This issue depends on multiple environment variables being set before being invoked:
+// D_CA_AGREE, expecting a value of "true" - checks to see if the user has agreed to the CA terms
+// D_CA_EMAIL, expecting an email address - this is used for the ACME account
+// D_DNS_PROVIDER, expecting a DNS provider, e.g. "cloudflare" - this is used for the DNS challenge
+// D_API_TOKEN, expecting an API token for the DNS provider - this is used to authenticate to the DNS provider
```

Comment:
Why use environment variables for this? Even if we decide to go that way, I'd say the main function or module should handle details like that, and this module should take these parameters in the `IssueCertificate` method or perhaps as fields of a struct you define. Doing it this way would make it harder to write tests on this module, because each test function would have to populate the environment appropriately.

Reply:
Yeah, I think that makes sense. I followed the design pattern that Lego had created, but I wasn't really happy with it.

This comes down to my question. How is the user going to execute this? Should it be a CLI based utility with flags? Should it be a configuration file?

Reply:
It seems like all these values now come via config file, so should this comment be updated?